### PR TITLE
Add tests for some edge cases to prevent regressions

### PR DESCRIPTION
**Pull request by llucax**
Add targeted tests adapted to the v1.x.x branch that verify behavior for issues identified in the pull request #1371:

- Data sourcing: Verify duplicate subscription and unknown components with same channel name does not hang. There is also a test for invalid metrics, but this one is currently failing (it hangs), but since we plan to remove the data sourcing actor, we delay the fix for now.

- Resampling: Verify invalid metric request does not block later valid request.

- Power status subscriptions (battery, EV charger, PV pools): Verify same-instance subscriptions share the registry channel correctly.

### CODE REVIEW

File: tests/actor/test_resampling.py [hunk 1]
```diff
@@ -184,3 +184,48 @@ async def test_duplicate_request(
         )
 
         await resampling_actor._resampler.stop()  # pylint: disable=protected-access
+
+
+async def test_stalled_request_does_not_block_later_subscriptions() -> None:
+    """Ensure resampling does not head-of-line block on v1.x.x."""
```
Comment (simonvoelcker) [inline tests/actor/test_resampling.py]: Can you explain the significance of the 1.x.x branch here? Was the behavior different on 0.x.x and/or do we keep backwards compatibility?

File: tests/timeseries/_pv_pool/test_pv_pool.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+from frequenz.channels import Broadcast
+from frequenz.client.common.microgrid.components import ComponentId
+from frequenz.quantities import Power
+from pytest_mock import MockerFixture
+
+from frequenz.sdk import timeseries
+from frequenz.sdk._internal._channels import ChannelRegistry
+from frequenz.sdk.microgrid._power_managing import ReportRequest, _Report
+from frequenz.sdk.timeseries.pv_pool._pv_pool import PVPool
```
Comment (simonvoelcker) [inline tests/timeseries/_pv_pool/test_pv_pool.py]:
```python
from frequenz.sdk.timeseries.pv_pool import PVPool
```

This should work, same in the line below and in the similar EV charger tests.

**Review by simonvoelcker — DISMISSED** [review]
Big fan of more tests 👍

File: tests/actor/test_resampling.py (hunk 1 above)
Comment (llucax) [reply tests/actor/test_resampling.py]: AI glitch, it was on comparison with #1371. Will remove.

File: tests/timeseries/_pv_pool/test_pv_pool.py (hunk 2 above)
Comment (llucax) [reply tests/timeseries/_pv_pool/test_pv_pool.py]: Damn, will remove other potential internal imports. Good catch!

**Comment by llucax** [comment]
[Fixed](https://github.com/frequenz-floss/frequenz-sdk-python/compare/83caed478e44fbb8bdef22ab878e8a05da45a44b..b32620a0a996729c3b5f53c8eeaad918a6785bbd).

**Review by simonvoelcker — APPROVED** [review]
